src/librustdoc/html/static/js/search.js

```diff
@@ -133,11 +124,337 @@ window.initSearch = function(rawSearchIndex) {
         searchState.input.value = params.search || "";
     }
 
+    function isWhitespace(c) {
+        return " \t\n\r".indexOf(c) !== -1;
+    }
+
+    function isSpecialStartCharacter(c) {
+        return "<\"".indexOf(c) !== -1;
+    }
+
+    function isEndCharacter(c) {
+        return ",>-".indexOf(c) !== -1;
+    }
+
+    function isStopCharacter(c) {
+        return isWhitespace(c) || isEndCharacter(c);
+    }
+
+    function isErrorCharacter(c) {
+        return "()".indexOf(c) !== -1;
```

Comment:
The eBNF is written with a specific list of allowed characters, while this is a list of banned ones. They don't match.

For example, consider the query `a'`, an alphanumeric followed by an apostrophe. According to that grammar, it should be rejected, but this JS parser thinks it's a valid ident.

Reply:
This is different in this case. Think about it as "reserved for potential future usage".

Reply:
Okay, that makes sense, but it still doesn't help with the overall problem. This parser is accepting `'` as a valid part of an identifier, but the formal grammar doesn't.

Reply:
Ah indeed! I'm just not sure how to handle the other cases actually... The biggest issue is that I don't know how to check for the letters (the unicode one more precisely). I'll need to investigate how to check if a character is in the accepted range we defined.

Reply:
Awesome, thanks!

Reply:
Since we don't handle unicode from the eBNF, I won't need it actually...